Assets/HoloToolkit/Input/Scripts/InputManager.cs

```diff
@@ -94,7 +94,14 @@ public void ClearModalInputStack()
         /// <param name="listener">Listener to add.</param>
         public void AddGlobalListener(GameObject listener)
         {
-            globalListeners.Add(listener);
+            if (!globalListeners.Contains(listener))
+            {
+                globalListeners.Add(listener);
+            }
+            else
+            {
+                Debug.LogWarningFormat("[{0}] This listener has already been added.", listener.name);
+            }
```

Comment:
Just for context: the reason I wasn't doing that was to avoid having to go through the queue every time we add something to it. That being said, I think the perf impact of doing this is minimal compared to the benefits, so this look OK to me